File: app/src/main/java/com/kacper/krakgo/screens/sign_in/register_part2/RegisterPart2Contract.java
package com.kacper.krakgo.screens.sign_in.register_part2;

import android.net.Uri;

import com.kacper.krakgo.model.UserDetails;

/**
 * Created by kacper on 27/10/2017.
 */

public interface RegisterPart2Contract {
    interface Presenter {
        void saveUserDetails(Uri mPhotoUri);
        void updateUserProfile(UserDetails userDetails);
    }

    interface View {
        void photoUploadFinished(Uri downloadUrl);
        void userDetailsUpdated();
        void showError(Exception e);
    }
}

File: app/src/main/java/com/kacper/krakgo/screens/sign_in/register_part2/RegisterPart2Contract.kt
package com.kacper.krakgo.screens.sign_in.register_part2

import android.net.Uri
import com.kacper.krakgo.model.UserDetails
import com.kacper.krakgo.mvp.MvpPresenter
import com.kacper.krakgo.mvp.MvpView

object RegisterPart2Contract {
    interface Presenter : MvpPresenter<RegisterPart2Contract.View>{
        fun saveUserDetails(mPhotoUri: Uri)
        fun updateUserProfile(userDetails: UserDetails)
    }

    interface View : MvpView{
        fun photoUploadFinished(downloadUrl: Uri)
        fun userDetailsUpdated()
        override fun showError(e: Exception)
    }
}